### ISSUE TITLE
Automated task creation, clashing keys

### ISSUE THREAD

**Issue by pape-io** [cider-ci/cider-ci#12]
3.5.0

```
ERROR: duplicate key value violates unique constraint "index_tasks_on_job_id_and_name"
  Detail: Key (job_id, name)=(3296398a-85fe-45df-b093-e270b23b6cee, front_end_logger_spec.rb) already exists.

  THROWABLE: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "index_tasks_on_job_id_and_name"
  Detail: Key (job_id, name)=(3296398a-85fe-45df-b093-e270b23b6cee, front_end_logger_spec.rb) already exists.[("cider_ci.builder.task$create_db_task.invoke (task.clj:34)" "cider_ci.builder.tasks$build_tasks$fn__8554.invoke (tasks.clj:131)" "cider_ci.builder.tasks$build_tasks.invoke (tasks.clj:129)" "cider_ci.builder.tasks$create_tasks.invoke (tasks.clj:145)" "cider_ci.builder.tasks$create_tasks_and_trials.invoke (tasks.clj:162)" "cider_ci.builder.jobs$invoke_create_tasks_and_trials.invoke (jobs.clj:64)" "cider_ci.builder.jobs$create.invoke (jobs.clj:79)" "cider_ci.builder.jobs.trigger$trigger_jobs$fn__8723$fn__8724.invoke (trigger.clj:107)" "cider_ci.builder.jobs.trigger$trigger_jobs$fn__8723.invoke (trigger.clj:101)" "cider_ci.builder.jobs.trigger$trigger_jobs.invoke (trigger.clj:100)" "cider_ci.builder.jobs.trigger$evaluate_branch_updated_message.invoke (trigger.clj:129)" "cider_ci.utils.messaging$create_handler$fn__8483.invoke (messaging.clj:98)")]
```

but specification looks okay: 

```
---
name: Tests
run-on:
- type: branch
  include-match: ^.*cider-ci.*$
context:
  subcontexts:
  - name: All rspec tests
    tasks:
      ...
      models/front_end_logger_spec.rb:
        environment-variables:
          CIDER_CI_TASK_FILE: spec/models/front_end_logger_spec.rb
      requests/front_end_logger_spec.rb:
        environment-variables:
          CIDER_CI_TASK_FILE: spec/requests/front_end_logger_spec.rb
      ...
    task-defaults:
      ports:
        CAPYBARA_PORT:
          max: 8999
          min: 8000
          inet_address: localhost
        PHANTOMJS_PORT:
          max: 44999
          min: 44600
          inet_address: localhost
      scripts:
        test:
          body: |
            #!/usr/bin/env bash
            bundle exec rspec $CIDER_CI_TASK_FILE
  task-defaults:
    scripts:
      test:
        start-when:
        - script: bundle
      bundle:
        body: bundle install
```

**Comment by DrTom** [comment]
There is some automatism to convert keys into names if no name is given. 
It is most likely a bug. But can you try the following workaround: give each of the two tasks a uniq name via the `name:` property. 

```
  models/front_end_logger_spec.rb:
        name:  models/front_end_logger_spec
        environment-variables:
          CIDER_CI_TASK_FILE: spec/models/front_end_logger_spec.rb
      requests/front_end_logger_spec.rb:
        name: requests/front_end_logger_spec
        environment-variables:
          CIDER_CI_TASK_FILE: spec/requests/front_end_logger_spec.rb
```

**Comment by DrTom** [comment]
I had a look into the code and yes indeed this is bug. There are two ways to resolve it 
1. just lift the db constraint, 
2. change the code such that the full path is used and not only the filename. 

We should implement 2. because it is also confusing to have two tasks with the same name.

**Comment by DrTom** [comment]
This is fixed as of 3.7.O-RC.1 https://github.com/cider-ci/cider-ci/tree/Cider-CI_3.7.0-RC.1.
